app/src/test/java/app/envelop/ui/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,78 @@
+package app.envelop.ui.login
+
+import app.envelop.common.Operation
+import app.envelop.domain.LoginService
+import app.envelop.ui.common.Finish
+import app.envelop.ui.common.LoadingState
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Single
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+
+class LoginViewModelTest {
+
+    val loginService = mock<LoginService>()
+    val loginViewModel = LoginViewModel()
+
+    @Before
+    fun setUp() {
+        loginViewModel.loginService = loginService
+    }
+
+    @Test
+    fun isLoggingIn() {
```

Comment:
Ideally test names should convey more information about what they are testing. For example, in this case, you're testing if the `isLoggingIn` stream comes back to Idle after a successful login click.